ibis_omniscidb/__init__.py

```diff
@@ -1,108 +1,773 @@
+        )
+
+        # used for UDF
+        new_backend.udf = OmniSciDBUDF(
+            host=new_backend.con._host,
+            port=new_backend.con._port,
+            database=new_backend.con._dbname,
+            user=new_backend.con._user,
+            password=new_backend.con._password,
+        )
+        return new_backend
+
+    def close(self):
+        """Close OmniSciDB connection and drop any temporary objects."""
+        self.con.close()
+
+    def _adapt_types(self, descr):
+        names = []
+        adapted_types = []
+
+        for col in descr:
+            names.append(col.name)
+            col_type = OmniSciDBDataType._omniscidb_to_ibis_dtypes[col.type]
+            col_type.nullable = col.nullable
+            adapted_types.append(col_type)
+        return names, adapted_types
+
+    def _check_execution_type(
+        self, ipc: Optional[bool], gpu_device: Optional[int]
+    ):
+        """
+        Check if the execution type (ipc and gpu_device) is valid.
+
+        Parameters
+        ----------
+        ipc : bool, optional
+        gpu_device : int, optional
+
+        Raises
+        ------
+        com.IbisInputError
+            if "gpu_device" is not None and "ipc" is False
+        """
+        if gpu_device is not None and ipc is False:
+            raise com.IbisInputError(
+                'If GPU device is provided, IPC parameter should '
+                'be True or None (default).'
+            )
+
+    def _fully_qualified_name(self, name, database):
+        # OmniSciDB raises error sometimes with qualified names
+        return name
+
+    def _get_list(self, cur):
+        tuples = cur.cursor.fetchall()
+        return [v[0] for v in tuples]
+
+    def _get_schema_using_query(self, query):
+        result = self.raw_sql(query)
+        # resets the state of the cursor and closes operation
+        result.cursor.fetchall()
+        names, ibis_types = self._adapt_types(
+            _extract_column_details(result.cursor._result.row_set.row_desc)
+        )
+
+        return sch.Schema(names, ibis_types)
+
+    def _get_schema_using_validator(self, query):
+        result = self.con._client.sql_validate(self.con._session, query)
+        return sch.Schema.from_tuples(
+            (
+                r.col_name,
+                OmniSciDBDataType._omniscidb_to_ibis_dtypes[
+                    pyomnisci_dtype._VALUES_TO_NAMES[r.col_type.type]
+                ],
+            )
+            for r in result
+        )
+
+    def _get_table_schema(self, table_name, database=None):
+        """Get table schema.
+
+        Parameters
+        ----------
+        table_name : str
+        database : str
+
+        Returns
+        -------
+        schema : ibis Schema
+        """
+        table_name_ = table_name.split('.')
+        if len(table_name_) == 2:
+            database, table_name = table_name_
+        return self.get_schema(table_name, database)
+
+    def raw_sql(
```

Comment:
is this the `execute` method used previously ?

maybe I am not following you here but ..  

after reading the base class (https://github.com/ibis-project/ibis/blob/master/ibis/backends/base/sql/__init__.py#L66)
I guess is that we should keep it as `execute`
and have raw_sql just for raw sql execution.

Reply:
I got it. thanks for the explanation.

I think that raw_sql should receive just a string for query parameter. 
we can open an issue for that and work on that in a follow-up PR.

thanks @datapythonista